feat: add C implementation for `stats/base/dists/gumbel/median`

vivekmaurya001 opened the pull request:
Resolves #3649 

## Description

- adds C implementation for @stdlib/stats/base/dists/gumbel/median along with relevant tests, benchmarks and examples

This pull request:

- Progresses: https://github.com/stdlib-js/stdlib/issues/3649
## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)


* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

vivekmaurya001 committed:
feat(add c implementation): add stats/base/dists/gumbel/median

vivekmaurya001 committed:
Merge branch 'gumbel/median' of https://github.com/vivekmaurya001/stdlib into gumbel/median

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/gumbel/median/src/main.c:
```diff
@@ -0,0 +1,47 @@
+* // returns ~0.367
+*/
+double stdlib_base_dists_gumbel_median( const double mu, const double beta ) {
+	double LLN2 = stdlib_base_ln( STDLIB_CONSTANT_FLOAT64_LN2 );
```
Should move constant to outer scope as a `static const` to avoid recomputation.

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/gumbel/median/src/main.c:
```diff
@@ -0,0 +1,47 @@
+#include "stdlib/stats/base/dists/gumbel/median.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/ln.h"
+#include "stdlib/constants/float64/ln_two.h"
+
+
+/**
```
```suggestion
#include "stdlib/constants/float64/ln_two.h"

/**
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/gumbel/median/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,141 @@
+
+	for ( i = 0; i < 100; i++ ) {
+		mu[ i ] = random_uniform( -50.0, 50.0 );
+		beta[ i ] = random_uniform( 0.0, 20.0 ) + STDLIB_CONSTANT_FLOAT64_EPS;
```
```suggestion
		beta[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 20.0 );
```

Planeshifter reviewed (changes requested):
Looking good overall. We can optimize the implementation a little, but otherwise this should be ready to land.

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into gumbel/median

Planeshifter committed:
Merge branch 'develop' into gumbel/median

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
chore: minor clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter reviewed (approved):
LGTM after clean-up.
